internal/services/appconfiguration/app_configuration_replica_resource.go

```diff
@@ -0,0 +1,170 @@
+package appconfiguration
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/appconfiguration/2023-03-01/configurationstores"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/appconfiguration/2023-03-01/replicas"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.Resource = ReplicaResource{}
+
+type ReplicaResource struct{}
+
+func (r ReplicaResource) ModelObject() interface{} {
+	return &ReplicaResourceSchema{}
+}
+
+type ReplicaResourceSchema struct {
+	ConfigurationStoreId string `tfschema:"configuration_store_id"`
+	Endpoint             string `tfschema:"endpoint"`
+	Location             string `tfschema:"location"`
+	Name                 string `tfschema:"name"`
+	ResourceGroupName    string `tfschema:"resource_group_name"`
+}
+
+func (r ReplicaResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return replicas.ValidateReplicaID
+}
+func (r ReplicaResource) ResourceType() string {
+	return "azurerm_app_configuration_replica"
+}
+func (r ReplicaResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"configuration_store_id": {
+			ForceNew:     true,
+			Required:     true,
+			Type:         pluginsdk.TypeString,
+			ValidateFunc: configurationstores.ValidateConfigurationStoreID,
+		},
+		"location": commonschema.Location(),
+		"name": {
+			ForceNew:     true,
+			Required:     true,
+			Type:         pluginsdk.TypeString,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+	}
+}
+
+func (r ReplicaResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"endpoint": {
+			Computed: true,
+			Type:     pluginsdk.TypeString,
+		},
+	}
+}
+
+func (r ReplicaResource) Create() sdk.ResourceFunc {
```

Comment:
rather than being a separate resource, this likely should be embedded within the App Configuration resource as a `replica` block, in the same way we've done for CosmosDB, since when making changes to the primary it can be necessary to make changes to the replicas too?

Reply:
@teowa, it is a 1:1 mapping, so unless there is a reason it needs to be separate, ie permissions / policy need updating with the created resource id before replication, it should be within the parent resource. also, as tom mentioned it simplifies the situations when the primary/secondary needs to be updated before/after a change

Reply:
This should be 1:n mapping, the API supports creating 1 replica per azure location, so there may be multiple replicas under one app configuration.

Reply:
If we make the `replica` a inline block inside the azurerm_app_configuration, but in create/update if the replica fails to create but its parent resource is successfully created, we still need to write parent resource to state? if we don't write the parent resource into state, there will be `tf.ImportAsExistsError` error. So how can we write parial state under current design(we call read function to write the state only if the whole create/updaet successfully)?

Reply:
``` in create/update if the replica fails to create but its parent resource is successfully created, we still need to write parent resource to state```

we would fail creation then.

this is much like the cosmosdb resource with its replication locations property. As tom said this shouljd be part of the parent resource we should mimic the way cosmos db does it?